src/rpc/protocol.h

```diff
@@ -106,4 +106,19 @@ void DeleteAuthCookie();
 /** Parse JSON-RPC batch reply into a vector */
 std::vector<UniValue> JSONRPCProcessBatchReply(const UniValue &in, size_t num);
 
+class JSONRPCRequest
```

Comment:
I'm not 100% convinced that `protocol.h` is the right place for this architecturally, as it's supposed to be functions and data structures shared between RPC client and server, and `JSONRPCRequest` is internal to the server. I understand the rationale, though.

Reply:
I think you're right, but I also think we have stuff in `protocol` that does not belong there to begin with. I made an attempt at cleaning this up by splitting the 'request' stuff into a separate file. This moves stuff from `server.cpp` and `server.h` and `protocol.h` into `request.h`, and `protocol.cpp` is replaced by `request.cpp`.